engine/docker/docker.go

```diff
@@ -565,7 +598,15 @@ func (e *Docker) createNetworkWithRetries(ctx context.Context,
 		driver = "nat"
 	}
 
-	var err error
+	// Check if the network already exists
+	_, _, err := e.client.NetworkInspectWithRaw(ctx, pipelineConfig.Network.ID, types.NetworkInspectOptions{})
+	if err == nil {
+		return nil
+	} else if !client.IsErrNotFound(err) {
+		// If the error is not "network not found", return it
+		return err
+	}
+
```

Comment:
Won't this break CI ? We are not creating network and returning directly?

Reply:
It should not break CI as it will fall into 3rd case above